src/Core/Settings.h

```diff
@@ -143,6 +143,7 @@ class IColumn;
     \
     M(UInt64, min_bytes_to_use_direct_io, 0, "The minimum number of bytes for reading the data with O_DIRECT option during SELECT queries execution. 0 - disabled.", 0) \
     M(UInt64, min_bytes_to_use_mmap_io, (64 * 1024 * 1024), "The minimum number of bytes for reading the data with mmap option during SELECT queries execution. 0 - disabled.", 0) \
+    M(UInt64, max_bytes_to_use_mmap_io, 0, "The maximum number of bytes for reading the data with mmap option during SELECT queries execution. 0 - no limit.", 0) \
```

Comment:
Note, that right now it is disabled, but I would rather set it 1GiB (or similar)